ICSharpCode.Decompiler/DecompilerEventSource.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Tracing;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using ICSharpCode.Decompiler.TypeSystem;
+using ICSharpCode.Decompiler.TypeSystem.Implementation;
+using Mono.Cecil;
+
+namespace ICSharpCode.Decompiler
+{
+	[EventSource(Name = "DecompilerEventSource")]
+	public sealed class DecompilerEventSource : EventSource
+	{
+		public static DecompilerEventSource Log = new DecompilerEventSource();
```

Comment:
Does the log have to go through global state?
What if I'm a user of the ICS.Decompiler library and want decompile two assemblies in parallel while keeping the logs separate?